app/models/guide.rb

```diff
@@ -37,28 +39,114 @@ def search_data
     as_json only: [:name, :overview, :crop_id]
   end
 
+  def basic_needs
+    if !user.gardens.empty?
+      # We should probably store these in the DB
+      basic_needs = [{
+          name: 'Sun / Shade',
+          overlap: [],
+          total: [],
+          percent: 0,
+          user: user.gardens.first.average_sun
+        }, {
+          name: 'Location',
+          overlap: [],
+          total: [],
+          percent: 0,
+          user: user.gardens.first.type
+        }, {
+          name: 'Soil Type',
+          overlap: [],
+          total: [],
+          percent: 0,
+          user: user.gardens.first.soil_type
+        }  # , {
+        #   name: 'pH Range',
+        # }, {
+        #   name: 'Temperature'
+        # }, {
+        #   name: 'Water Use'
+        # }, {
+        #   name: 'Practices',
+        #   value: practices
+        # }, {
+        #   name: 'Time Commitment'
+        # }, {
+        #   name: 'Physical Ability'
+        # }, {
+        #   name: 'Time of Year'
+        # }
+      ]
+
+      find_overlap_in basic_needs
+    end
+  end
+
   def compatibility_score
-    # Make our random scores consistent based on the first character of the crop name
-    # srand(name[0].ord)
-    # rand(100);
-    nil
+    if !user.gardens.empty?
+      count = 0
+      sum = basic_needs.inject(0) do |memo, n|
+        count += 1
+        n[:percent] ? memo + n[:percent] : memo
+      end
+      (sum.to_f / count * 100).round
+    end
   end
 
   def compatibility_label
-    ''
-    # TODO:
-    # score = compatibility_score
-
-    # if score.nil?
-    #   return ''
-    # elsif score > 75
-    #   return 'high'
-    # elsif score > 50
-    #   return 'medium'
-    # else
-    #   return 'low'
-    # end
+    score = compatibility_score
+
+    if score.nil?
+      return ''
+    elsif score > 75
+      return 'high'
+    elsif score > 50
+      return 'medium'
+    else
+      return 'low'
+    end
   end
 
-  slug :name
+  private
+
+  # For each stage, find the overlapping needs,
+  # and then calculate the percentage overlap of each need
+  # ToDO this can be cleaned up, probably significantly
+  # by externalizing the basic_need structure.
+  def find_overlap_in(basic_needs)
+    stages.each do |stage|
+      basic_needs.each do |need|
+        # This is bad structure
+        if need[:name] == "Sun / Shade"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.